File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/ImageApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class ImageApi {
    private final HttpClient client;
    
    public ImageApi(HttpClient client) {
        this.client = client;
    }

    /** Update an existing image */
    public PlusApiResultPlusImageVO update(PlusImageForm body) throws Exception {
        return (PlusApiResultPlusImageVO) client.put(ApiPaths.backendPath("/image"), body);
    }

    /** Create a new image */
    public PlusApiResultPlusImageVO create(PlusImageForm body) throws Exception {
        return (PlusApiResultPlusImageVO) client.post(ApiPaths.backendPath("/image"), body);
    }

    /** Get images by page */
    public PlusApiResultPagePlusImageVO listByPage(QueryListForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusImageVO) client.post(ApiPaths.backendPath("/image/list"), body, params);
    }

    /** Get all images */
    public PlusApiResultListPlusImageVO listAllEntities(QueryListForm body) throws Exception {
        return (PlusApiResultListPlusImageVO) client.post(ApiPaths.backendPath("/image/list/all"), body);
    }

    /** Get an image by ID */
    public PlusApiResultPlusImageVO getById(String id) throws Exception {
        return (PlusApiResultPlusImageVO) client.get(ApiPaths.backendPath("/image/" + id + ""));
    }

    /** Delete an image */
    public PlusApiResultBoolean delete(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/image/" + id + ""));
    }
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/ImageApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class ImageApi(private val client: HttpClient) {

    /** Update an existing image */
    suspend fun update(body: PlusImageForm): PlusApiResultPlusImageVO? {
        return client.put(ApiPaths.backendPath("/image"), body) as? PlusApiResultPlusImageVO
    }

    /** Create a new image */
    suspend fun create(body: PlusImageForm): PlusApiResultPlusImageVO? {
        return client.post(ApiPaths.backendPath("/image"), body) as? PlusApiResultPlusImageVO
    }

    /** Get images by page */
    suspend fun listByPage(body: QueryListForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusImageVO? {
        return client.post(ApiPaths.backendPath("/image/list"), body, params) as? PlusApiResultPagePlusImageVO
    }

    /** Get all images */
    suspend fun listAllEntities(body: QueryListForm? = null): PlusApiResultListPlusImageVO? {
        return client.post(ApiPaths.backendPath("/image/list/all"), body) as? PlusApiResultListPlusImageVO
    }

    /** Get an image by ID */
    suspend fun getById(id: String): PlusApiResultPlusImageVO? {
        return client.get(ApiPaths.backendPath("/image/$id")) as? PlusApiResultPlusImageVO
    }

    /** Delete an image */
    suspend fun delete(id: String): PlusApiResultBoolean? {
        return client.delete(ApiPaths.backendPath("/image/$id")) as? PlusApiResultBoolean
    }
}
